lua/neogit/lib/git/repository.lua

```diff
@@ -21,6 +21,10 @@ local function empty_state()
       branch = nil,
       commit_message = nil
     },
+    tag = {
+      name = nil,
+      distance = nil,
+    },
```

Comment:
Since it's always using `HEAD`, I think this could be pulled into the `head` table, ie:

```lua
{ head = { tag = { name = nil, distance = nil } } }
```